src/utils/ISoilUtils.C

```diff
@@ -0,0 +1,244 @@
+    for (std::size_t j = 0; j < number_of_points; ++j)
+    {
+      backbone_strain[i][j] = pow(10.0, -6.0 + 5.0 / (number_of_points - 1) * j);
+      Real theta =
+          theta_1[i] +
+          theta_2[i] * theta_4[i] *
+              pow(backbone_strain[i][j] / (taumax[i] / initial_shear_modulus[i]), theta_5[i]) /
+              (pow(theta_3[i], theta_5[i]) +
+               pow(backbone_strain[i][j] / (taumax[i] / initial_shear_modulus[i]), theta_5[i]));
+      modulus[i][j] =
+          (theta == 0)
+              ? 1.0 / (1.0 + backbone_strain[i][j] / (taumax[i] / initial_shear_modulus[i]))
+              : (1.0 / (backbone_strain[i][j] / (taumax[i] / initial_shear_modulus[i]))) *
+                    (0.5 / theta *
+                     (1.0 + backbone_strain[i][j] / (taumax[i] / initial_shear_modulus[i]) -
+                      sqrt(pow(1.0 + backbone_strain[i][j] / (taumax[i] / initial_shear_modulus[i]),
+                               2.0) -
+                           4.0 * theta * backbone_strain[i][j] /
+                               (taumax[i] / initial_shear_modulus[i]))));
+      backbone_stress[i][j] = modulus[i][j] * initial_shear_modulus[i] * backbone_strain[i][j];
+    }
+  }
+}
+
+void
+ISoilUtils::computeCoulombBackbone(std::vector<std::vector<Real>> & backbone_stress,
+                                   std::vector<std::vector<Real>> & backbone_strain,
+                                   const std::vector<unsigned int> & _layer_ids,
+                                   std::vector<Real> & initial_shear_modulus,
+                                   std::vector<Real> & friction_coefficient,
+                                   std::vector<Real> & hardening_ratio,
+                                   const std::vector<Real> & _p_ref,
+                                   std::string name)
+{
+  backbone_strain.resize(_layer_ids.size());
+  backbone_stress.resize(_layer_ids.size());
+  for (std::size_t i = 0; i < _layer_ids.size(); i++)
+  {
+    backbone_strain[i].resize(2);
+    backbone_stress[i].resize(2);
+    if (hardening_ratio[i] >= 1.0)
+      mooseError("Error in " + name + ". hardening_ratio cannot be greater than or equal to 1.0.");
+    backbone_strain[i][0] = _p_ref[i] * friction_coefficient[i] / initial_shear_modulus[i];
+    backbone_stress[i][0] = _p_ref[i] * friction_coefficient[i];
+    backbone_strain[i][1] = 100.0;
+    backbone_stress[i][1] = backbone_stress[i][0] +
+                            (backbone_strain[i][1] - backbone_strain[i][0]) * hardening_ratio[i] *
+                                initial_shear_modulus[i];
+  }
+}
+
+void
+ISoilUtils::computeSoilLayerProperties(
+    std::vector<std::vector<Real>> & backbone_stress,
+    std::vector<std::vector<Real>> & backbone_strain,
+    const std::vector<unsigned int> & _layer_ids,
+    std::vector<std::vector<Real>> & _youngs,
+    std::vector<std::vector<Real>> & _yield_stress, // *** YIELD STRAIN, NOT YIELD STRESS***
```

Comment:
remove comment and call this yield strain.

Reply:
Remove the leading underscore from all these variables. Also, all but youngs and yeild_stress should be const

Reply:
I am a little bit nervous about changing this since this is @sveerara 's code and she understands it much better than I do. She had a reason to call this yield_stress, but I didn't entirely understand.